2214/2/popular_words_counter.rb

```diff
@@ -0,0 +1,32 @@
+require_relative 'word_with_quantity'
+require_relative 'battler'
+
+class PopularWordsCounter
+  PREPOSITIONS_FILE = 'Предлоги'.freeze
+  def initialize(specific_battler, top_words)
+    @battles = specific_battler.battles
+    @top_words = top_words
+  end
+
+  def count
+    popular_words.first(@top_words).map { |popular_word| puts "#{popular_word.word} - #{popular_word.quantity} раз" }
+  end
+
+  private
+
+  def popular_words
+    usefull_words.uniq.map { |word| WordWithQuantity.new(word, usefull_words.count(word)) }.sort_by(&:quantity).reverse
```

Comment:
Please use memoization in methods like `#popular_words` and `#usefull_words` and other because they are costly to run. They are relatively slow.

Please measure the execution time for the current version and then for the version of the app with memoization and share it in the chat with me.